File: clients/rust/src/api/rate_limiter.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

#[derive(Default)]
pub struct RateLimiterLimitOptions {
    pub idempotency_key: Option<String>,
}

#[derive(Default)]
pub struct RateLimiterGetRemainingOptions {
    pub idempotency_key: Option<String>,
}

pub struct RateLimiter<'a> {
    cfg: &'a Configuration,
}

impl<'a> RateLimiter<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Rate Limiter Check and Consume
    pub async fn limit(
        &self,
        rate_limiter_check_in: RateLimiterCheckIn,
        options: Option<RateLimiterLimitOptions>,
    ) -> Result<RateLimiterCheckOut> {
        let RateLimiterLimitOptions { idempotency_key } = options.unwrap_or_default();

        crate::request::Request::new(http::Method::POST, "/api/v1/rate-limiter/limit")
            .with_optional_header_param("idempotency-key", idempotency_key)
            .with_body_param(rate_limiter_check_in)
            .execute(self.cfg)
            .await
    }

    /// Rate Limiter Get Remaining
    pub async fn get_remaining(
        &self,
        rate_limiter_get_remaining_in: RateLimiterGetRemainingIn,
        options: Option<RateLimiterGetRemainingOptions>,
    ) -> Result<RateLimiterGetRemainingOut> {
        let RateLimiterGetRemainingOptions { idempotency_key } = options.unwrap_or_default();

        crate::request::Request::new(http::Method::POST, "/api/v1/rate-limiter/get-remaining")
            .with_optional_header_param("idempotency-key", idempotency_key)
            .with_body_param(rate_limiter_get_remaining_in)
            .execute(self.cfg)
            .await
    }
}

File: clients/python/diom/apis/rate_limiter.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    RateLimiterCheckIn,
    RateLimiterCheckOut,
    RateLimiterGetRemainingIn,
    RateLimiterGetRemainingOut,
)
from .rate_limiter_namespace import (
    RateLimiterNamespace,
    RateLimiterNamespaceAsync,
)


class RateLimiterAsync(ApiBase):
    @property
    def namespace(self) -> RateLimiterNamespaceAsync:
        return RateLimiterNamespaceAsync(self._client)

    async def limit(
        self,
        rate_limiter_check_in: RateLimiterCheckIn,
    ) -> RateLimiterCheckOut:
        """Rate Limiter Check and Consume"""
        body = rate_limiter_check_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/rate-limit/limit",
            body=body,
            response_type=RateLimiterCheckOut,
        )

    async def get_remaining(
        self,
        rate_limiter_get_remaining_in: RateLimiterGetRemainingIn,
    ) -> RateLimiterGetRemainingOut:
        """Rate Limiter Get Remaining"""
        body = rate_limiter_get_remaining_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/rate-limit/get-remaining",
            body=body,
            response_type=RateLimiterGetRemainingOut,
        )


class RateLimiter(ApiBase):
    @property
    def namespace(self) -> RateLimiterNamespace:
        return RateLimiterNamespace(self._client)

    def limit(
        self,
        rate_limiter_check_in: RateLimiterCheckIn,
    ) -> RateLimiterCheckOut:
        """Rate Limiter Check and Consume"""
        body = rate_limiter_check_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1/rate-limit/limit",
            body=body,
            response_type=RateLimiterCheckOut,
        )

    def get_remaining(
        self,
        rate_limiter_get_remaining_in: RateLimiterGetRemainingIn,
    ) -> RateLimiterGetRemainingOut:
        """Rate Limiter Get Remaining"""
        body = rate_limiter_get_remaining_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1/rate-limit/get-remaining",
            body=body,
            response_type=RateLimiterGetRemainingOut,
        )
